src/exercises/Exercises.js

```diff
@@ -230,9 +284,14 @@ export default function Exercises({
 
     if (newIndex === exerciseSession.length) {
       setFinished(true);
+      LocalStorage.incrementAudioExperimentNoOfSessions();
+      LocalStorage.checkAudioExperimentCompleted();
+      api.logReaderActivity(api.AUDIO_EXP, articleID, "Session no: " + LocalStorage.getAudioExperimentNoOfSessions(), source);
+      if (LocalStorage.checkAudioExperimentCompleted()) {
```

Comment:
here you see the limitations of the `check...Completed` function name.
separate behavior in two functions: `is...Completed` and `set...Completed`. Then use them accordingly.